public/js/NeighbourImages.js

```diff
@@ -6,14 +6,26 @@ var _ = require('lodash');
  * @constructor
  */
 function NeighbourImages() {
-    var self = this;
-
     this.neighbours = [];
     this.selectedImage = "";
     this.neighbourImages = [];
 
-    var divWidth = Math.round($('#image-column').width()) - 20;
-    var imgWidth = Math.round(divWidth/3) - 20;
+    // cache jquery selectors used frequency by gallery methods
+    // call these in the constructor so the DOM is traversed only once
+    // before caching.
```

Comment:
Whenever you use ``$('#my-element')``, you're basically telling jQuery "hey, go search the entire DOM for ``#my-element``. When you save the call to jQuery in a variable, you're saying "hey, search the DOM for ``#my-element`` then remember where it was so you don't need to search next time". 

Previously I was (dumbly) caching the selector in the methods instead of the constructor, so each time the method was called.. so it was searching the DOM every time the method was called. Now the selectors are object properties, they're cached for the entire life of the object!

Caching your jQuery selectors is a really good practice when you're making multiple calls to the same DOM element, and you don't expect the structure of the DOM to change. Which is exactly the perfect use case for the image gallery. :)